### DISCUSSION TITLE
Is it posible to combine `useSetAtom` with `useReducerAtom`?

### DISCUSSION THREAD

**Discussion by Stainless2k** [pmndrs/jotai#2464] [Q&A] [answered]
I basically want a version of `useReducerAtom` that only returns the `dispatch`, for the same reasons `setAtom` exist, to reduce unnecessary re-renders.

In our app we use an `atomFamily` of `atomWithStorage` with the `useReducerAtom` hook, but at some places we don't care about the value we just need the `dispatch`

**Comment by dai-shi** [comment]
In that case, you want to simply use useSetAtom and create a custom hook to apply a reducer.
That made me think that we should rather deprecate `useReducerAtom`. I don't think it's widely used, and it's limited as you suggest.

**Reply by dai-shi** [reply, marked as answer]
#2467

**Reply by Stainless2k** [reply]
Thank you for the fast reply

**Reply by lsancho** [reply]
Hey @Stainless2k, could you please provide the solution you ended up with as an example?

**Reply by Stainless2k** [reply]
> Hey @Stainless2k, could you please provide the solution you ended up with as an example?

i used the recipe from the docs https://jotai.org/docs/recipes/use-reducer-atom
